src/router/index.tsx

```diff
@@ -41,16 +57,31 @@ const routerChildren = [
       },
     ],
   },
+  { path: "/api/auth/oauth2/kakao", element: <KaKaoRedirection />, auth: false },
 ];
 
+const browserRouter = routerChildren.map(({ path, element, auth, children }) => {
+  return {
+    path,
+    element: auth ? <RequireLoginLayout>{element}</RequireLoginLayout> : element,
+    children: children
+      ? children.map((child) => ({
```

Comment:
이렇게 복잡한 삼항연산자를 줄여볼 수도 있을 것 같아요!

```suggestion
children: children?.map((child) => ({
      path: child.path,
      element: auth ? <RequireLoginLayout>{child.element}</RequireLoginLayout> : child.element,
    })),
```